Automattic/jetpack
# Main Dashboard: improve performance by loading React dash only when connected / in dev mode

Improved the loading performance of the connection page my making sure that we don't load the admin.js file. 

This results in a much faster appearance of the connection button the jetpack dashboard page. 

Before:
https://cloudup.com/csvrz9kD32W

After:
https://cloudup.com/cvBU_z__CQa

**Notes**

This makes use of a React portal to hook 2 extra links to the footer, now relying on PHP for all links but those 2.

This introduces 2 issues:
1. It breaks tracking on the footer links.
2. It removes the "Reset" and "Dev Tools" buttons when not connected to WordPress.com / not in dev mode.

#### Testing instructions:
* Activate the jetpack plugin. Notice that the banner shows up as before.  
* Navigate to the Jetpack Dashboard. Notice that the page loads noticeably faster.
* Also test when in development mode, and in different connection scenarios. 

#### Proposed changelog entry for your changes:
* Improves the loading of the dashboard when Jetpack is not connected.
